### PR TITLE
Add networking flag option for cilium in eni mode

### PR DESCRIPTION
**Pull request by olemarkus**
Ref #14575

### CODE REVIEW

File: cmd/kops/integration_test.go [hunk 1]
```diff
@@ -516,6 +516,13 @@ func TestPrivateCiliumAdvanced(t *testing.T) {
 		runTestTerraformAWS(t)
 }
 
+func TestPrivateCiliumEni(t *testing.T) {
+	newIntegrationTest("privatecilium.example.com", "privateciliumeni").
```
Comment (johngmyers) [inline cmd/kops/integration_test.go:520]: perhaps a `-` before `eni` in the direectory name?

**Comment by johngmyers** [comment]
This would be easier to review if the update-expected.sh changes were in a separate commit.

File: upup/pkg/fi/cloudup/new_cluster.go [hunk 2]
```diff
@@ -1141,6 +1141,8 @@ func setupNetworking(opt *NewClusterOptions, cluster *api.Cluster) error {
 	case "cilium-etcd":
 		addCiliumNetwork(cluster)
 		cluster.Spec.Networking.Cilium.EtcdManaged = true
+	case "cilium-eni":
+		addCiliumNetwork(cluster)
```
Comment (johngmyers) [inline upup/pkg/fi/cloudup/new_cluster.go:1145]: It doesn't look like this is actually enabling ENI mode.

**Comment by johngmyers** [comment]
I think we should make sure that `--networking=cilium-eni --ipv6` works and behaves the same as `--networking=cilium --ipv6`.

File: upup/pkg/fi/cloudup/new_cluster.go (hunk 2 above)
Comment (olemarkus) [reply upup/pkg/fi/cloudup/new_cluster.go:1145]: Heh. I missed a commit as I wanted the test diff to be more visible. Added now.

**Comment by olemarkus** [comment]
> I think we should make sure that `--networking=cilium-eni --ipv6` works and behaves the same as `--networking=cilium --ipv6`.

We don't support eni mode for ipv6. That's already blocked by validation. I assume there would be a difference in behavior between the two: that cilium-operator does the prefix delegation and node IPAM rather than kops-controller. This support wasn't added until 1.12 though, I believe.

File: cmd/kops/integration_test.go [hunk 3]
```diff
@@ -510,6 +510,13 @@ func TestPrivateCiliumAdvanced(t *testing.T) {
 		runTestTerraformAWS(t)
 }
 
+func TestPrivateCiliumEni(t *testing.T) {
```
Comment (johngmyers) [inline cmd/kops/integration_test.go:513]:
```suggestion
func TestPrivateCiliumENI(t *testing.T) {
```

**Review by hakman — APPROVED** [review]

**Pull request merged**